### PR TITLE
Remove INT96 timestamps in cuDF Parquet writer

### LINKED ISSUE #10438: [BUG] Incorrect statistics for int96 timestamps in parquet
**Describe the bug**
The statistics calculation is not working properly when dealing with int96 timestamp types. Adding statistics checks in `test_parquet_writer_int96_timestamps` will cause pytest failures.

**Steps/Code to reproduce bug**
At the end of `test_parquet_writer_int96_timestamps`, adding the following code:
```python
    # Read back from pyarrow
    pq_file = pq.ParquetFile(gdf_fname)
    # verify each row group's statistics
    for rg in range(0, pq_file.num_row_groups):
        pd_slice = pq_file.read_row_group(rg).to_pandas()

        # statistics are per-column. So need to verify independently
        for i, col in enumerate(pd_slice):
            stats = pq_file.metadata.row_group(rg).column(i).statistics

            actual_min = cudf.Series(pd_slice[col].explode().explode()).min()
            stats_min = stats.min
            assert normalized_equals(actual_min, stats_min)

            actual_max = cudf.Series(pd_slice[col].explode().explode()).max()
            stats_max = stats.max
            assert normalized_equals(actual_max, stats_max)
```
The statistics test will fail.

**Expected behavior**
The test should pass.

**Additional context**
Not sure if the issue is in libcudf or in the python layer. Adding cpp unit tests for parquet statistics can help track down the issue.

### LINKED ISSUE #15847: [FEA] Support `arrow:Schema` in Parquet writer for faithful roundtrip with Arrow via Parquet
Support writing `arrow:Schema` to parquet file footer to allow faithful interop with Arrow via Parquet. The writer's side may be much more complicated and involve supporting all types, and computing and writing dictionary ids.

Related to: #15617

```[tasklist]
### Tasks
- [ ] #15875
- [ ] #15901
```

### PR DESCRIPTION
**Pull request by mhaseeb123**
## Description

Partially closes #15847.
Closes #10438 

This PR remonves INT96 timestamps from cuDF in accordance with Arrow. For backward compatibility and robustness, the Parquet reader still has the capability to read INT96 timestamps and convert to INT64. For more discussion, see #15875.

## Discussion
- [ ] Would Spark functionality or tests be affected by this change? if yes, we can postpone this PR. (@nvdbaranec)

## Answered
- [x] Following @etseidl's suggestion, PQ reader's capability to read and convert INT96 timestamps has been reverted to allow reading older parquet files with INT96 timestamp data.

CC @vuule @etseidl @nvdbaranec @GregoryKimball @galipremsagar for vis.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/include/cudf/io/parquet_metadata.hpp [hunk 1]
```diff
@@ -46,7 +46,6 @@ enum class TypeKind : int8_t {
   BOOLEAN              = 0,
   INT32                = 1,
   INT64                = 2,
-  INT96                = 3,  // Deprecated
```
Comment (etseidl) [inline cpp/include/cudf/io/parquet_metadata.hpp:49]: This should not be removed from the enum as it is still in the official spec.

File: cpp/src/io/parquet/decode_fixed.cu [hunk 2]
```diff
@@ -193,8 +193,6 @@ __device__ inline void gpuDecodeValues(
             }
             break;
         }
-      } else if (dtype == INT96) {
```
Comment (etseidl) [inline cpp/src/io/parquet/decode_fixed.cu:196]: I think decoding of int96 should remain possible, just remove the write path.

File: cpp/src/io/parquet/parquet_common.hpp [hunk 3]
```diff
@@ -34,7 +34,6 @@ enum Type : int8_t {
   BOOLEAN              = 0,
   INT32                = 1,
   INT64                = 2,
-  INT96                = 3,  // Deprecated
```
Comment (etseidl) [inline cpp/src/io/parquet/parquet_common.hpp:37]: same comment here...this should stay

**Review by etseidl — COMMENTED** [review]
Looks more like "Remove INT96 timestamps" than "Deprecate" 😅. I think it's fine to remove from the write path, but think cudf should still be capable of reading it, even if it's always just turned into a timestamp_ns.

**Comment by etseidl** [comment]
Thanks for this (IMO long overdue) update! Looks good. 👍

File: cpp/src/io/parquet/parquet_common.hpp (hunk 3 above)
Comment (mhaseeb123) [reply cpp/src/io/parquet/parquet_common.hpp]: Thanks for the valuable suggestions, Ed. Reverted!

File: cpp/include/cudf/io/parquet_metadata.hpp (hunk 1 above)
Comment (mhaseeb123) [reply cpp/include/cudf/io/parquet_metadata.hpp]: Reverted!

File: cpp/src/io/parquet/decode_fixed.cu (hunk 2 above)
Comment (mhaseeb123) [reply cpp/src/io/parquet/decode_fixed.cu]: Reverted!

**Comment by mhaseeb123** [comment]
Hi @nvdbaranec, I wanted to bring your attention to the changes in this PR (removes INT96 timestamps in cuDF) and wanted to ask if it would create any waves across Spark functionality or testing?

**Comment by mhaseeb123** [comment]
Closing the branch as we need INT96 timestamps for Spark.

**Pull request closed without merging**